operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -146,9 +151,206 @@ private void addInitContainer(Deployment baseDeployment, List<String> extensions
         baseDeployment.getSpec().getTemplate().getSpec().setInitContainers(Collections.singletonList(initContainer));
     }
 
+    public void validatePodTemplate(KeycloakStatusBuilder status) {
+        if (keycloakCR.getSpec() == null ||
+                keycloakCR.getSpec().getUnsupported() == null ||
+                keycloakCR.getSpec().getUnsupported().getPodTemplate() == null) {
+            return;
+        }
+        var overlayTemplate = this.keycloakCR.getSpec().getUnsupported().getPodTemplate();
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getName() != null) {
+            status.addWarningMessage("The name of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getNamespace() != null) {
+            status.addWarningMessage("The namespace of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getSpec() != null &&
+            overlayTemplate.getSpec().getContainers() != null &&
+            overlayTemplate.getSpec().getContainers().get(0) != null &&
+            overlayTemplate.getSpec().getContainers().get(0).getName() != null) {
+            status.addWarningMessage("The name of the keycloak container cannot be modified");
+        }
+    }
+
+    private <T, V> void mergeMaps(Map<T, V> map1, Map<T, V> map2, Consumer<Map<T, V>> consumer) {
+        var map = new HashMap<T, V>();
+        Optional.ofNullable(map1).ifPresent(e -> map.putAll(e));
+        Optional.ofNullable(map2).ifPresent(e -> map.putAll(e));
```

Comment:
Let's remember to discuss this point offline.

I do believe that this is perfectly idiomatic Java 11 code and I don't find it hard to read at all.
On the contrary keeping `null` checks around should be considered bad practice.

Over-engineering is about using higher level and un-needed abstractions, not for using Standard library constructs for their intended purpose.

Reply:
Per my understanding, over-engineering is a [synonym for unnecessarily complicated](https://www.collinsdictionary.com/dictionary/english/over-engineered). 😉

Yes, this is truly a Standard library construct and it certainly has its place when e.g. designing contracts. And yes, relying on `null` checks in contracts is considered a bad practice – I agree.

However, in this specific case we are really over-using it. We're not designing a contract here. We're replacing a super simple `if` with a more complex structure with lambda, which I see zero benefit in. It almost feels like we're using shiny new-ish features just for the sake of using it. 😉

As I said, no big deal here. But I'd like to avoid over-engineering/over-complicating (however you want to call it) stuff in future PRs.

Reply:
In this context using `Optional` is simply a safe and elegant expression for mapping the underlying concept 🙂 .
I would like to know more regarding confining `Optional` to contracts and the reasoning behind this practice, never heard about it.

I do not agree that a lambda expression is more "complex" than imperative code in any sense, it might be less "familiar" to you, but that's a completely different story.

> But I'd like to avoid over-engineering/over-complicating (however you want to call it) stuff in future PRs.

I do believe that we should agree on a common style __as a team__.

Reply:
I don't really want to go deeper into (insulting) discussions around Java experience. Let's stop here. 🙂 The last thing I'll say to this is this:
![IMAGE 2022-03-01 14:01:57](https://user-images.githubusercontent.com/13906855/156173581-db5e9fc6-f744-42a2-aef7-1e31a403b2eb.jpg)